src/controllers/NotificationController.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * Tests for Notification Controllers
+ * Send HTTP requests to Notification Controllers and check mocked responses
+ */
+
+import * as express from "express";
+import * as NotificationAPI from "../api/services/NotificationAPI";
+import { NotificationSubscriptionResponse } from "../api/types/NotificationSubscriptionResponse";
+import { PagoPaConfig } from "../Configuration";
+import { ControllerError } from "../enums/ControllerError";
+import { HttpErrorStatusCode } from "../enums/HttpErrorStatusCode";
+import { NotificationSubscriptionRequestType } from "../enums/NotificationSubscriptionType";
+import { FiscalCode } from "../types/FiscalCode";
+import * as AppResponseConverter from "../utils/AppResponseConverter";
+import * as RestfulUtils from "../utils/RestfulUtils";
+
+// Notification Controller
+export function notificationActivation(
+  req: express.Request,
+  res: express.Response,
+  pagoPaConfig: PagoPaConfig
+): void {
+  updateSubscription(
+    req,
+    res,
+    NotificationSubscriptionRequestType.ACTIVATION,
+    pagoPaConfig
+  );
+}
+
+export function notificationDeactivation(
+  req: express.Request,
+  res: express.Response,
+  pagoPaConfig: PagoPaConfig
+): void {
+  updateSubscription(
+    req,
+    res,
+    NotificationSubscriptionRequestType.DEACTIVATION,
+    pagoPaConfig
+  );
+}
+
+// Update user subscription to Notification Service
+function updateSubscription(
+  req: express.Request,
+  res: express.Response,
+  requestType: NotificationSubscriptionRequestType,
+  pagoPaConfig: PagoPaConfig
+): void {
+  // Check input
+  const fiscalCode = FiscalCode.decode(req.params.fiscalCode);
+  if (fiscalCode.isLeft()) {
+    RestfulUtils.sendErrorResponse(
+      res,
+      ControllerError.ERROR_INVALID_INPUT,
+      HttpErrorStatusCode.BAD_REQUEST
+    );
+    return;
+  }
+
+  // Require subscription to API
+  NotificationAPI.updateSubscription(
+    fiscalCode.value,
+    requestType,
+    res,
+    pagoPaConfig,
+    RestfulUtils.sendUnavailableAPIError,
+    (
```

Comment:
when dealing with async code, always prefer to use **promises** instead of this callback style 
ie. the code could become:

```ts
const result = await NotificationAPI.updateSubscription(...);
< do something with result >
```